### ISSUE TITLE
Swift UI Previews are crashing in Xcode 15.3 (worked fine in 15.2).

### ISSUE THREAD

**Issue by videolist** [realm/realm-swift#8518]
### How frequently does the bug occur?

Always

### Description

Swift UI previews started crashing somewhere in Realm when initializing the configuration. We are using in-memory Realm for previews. Used to work before upgrading to Xcode 15.3

### Stacktrace & log output

```shell
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Incident Identifier: 5F489663-385A-4AA3-BE9E-05CC06931387
CrashReporter Key:   54FBA5B8-80CA-4028-04C0-C0F05C4D7D20
Hardware Model:      MacBookPro18,2
Process:             Tezza [3963]
Path:                /Users/USER/Library/Developer/Xcode/UserData/Previews/Simulator Devices/D932BB63-490F-40AA-88FD-38183AFE80FD/data/Containers/Bundle/Application/180A7B93-D16E-4125-8FE1-F71B9D58EE05/Tezza.app/Tezza
Identifier:          com.tezzanyc.Tezza
Version:             8.4 (2152)
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd_sim [2460]
Coalition:           com.apple.CoreSimulator.SimDevice.D932BB63-490F-40AA-88FD-38183AFE80FD [1383]
Responsible Process: SimulatorTrampoline [2055]

Date/Time:           2024-03-13 14:28:58.5303 -0700
Launch Time:         2024-03-13 14:28:57.0829 -0700
OS Version:          macOS 14.4 (23E214)
Release Type:        User
Report Version:      104

Exception Type:  EXC_BAD_ACCESS (SIGSEGV)
Exception Subtype: KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes: 0x0000000000000001, 0x0000000000000000
VM Region Info: 0 is not in any region.  Bytes before following region: 4301783040
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      100680000-1025d8000    [ 31.3M] r-x/r-x SM=COW  /Users/USER/Library/Developer/Xcode/UserData/Previews/Simulator Devices/D932BB63-490F-40AA-88FD-38183AFE80FD/data/Containers/Bundle/Application/180A7B93-D16E-4125-8FE1-F71B9D58EE05/Tezza.app/Tezza
Termination Reason: SIGNAL 11 Segmentation fault: 11
Terminating Process: exc handler [3963]

Triggered by Thread:  0

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libswiftCore.dylib            	       0x11b8966d8 swift::TargetMetadata<swift::InProcess>::getGenericArgs() const + 20
1   libswiftCore.dylib            	       0x11b8be2c8 instantiateWitnessTable(swift::TargetMetadata<swift::InProcess> const*, swift::TargetProtocolConformanceDescriptor<swift::InProcess> const*, void const* const*, void**) + 140
2   libswiftCore.dylib            	       0x11b8b605c swift::_getWitnessTable(swift::TargetProtocolConformanceDescriptor<swift::InProcess> const*, swift::TargetMetadata<swift::InProcess> const*, void const* const*) + 4360
3   Tezza                         	       0x100ef59e0 lazy protocol witness table accessor for type LinkingObjects<Project> and conformance LinkingObjects<A> + 72
4   Tezza                         	       0x10127bd60 AudioItem.init() + 956 (AudioItem.swift:26)
5   Tezza                         	       0x10127c16c @objc AudioItem.init() + 20
6   Tezza                         	       0x1018f0a84 @nonobjc RLMObjectBase.init() + 24
7   Tezza                         	       0x1018ef2e4 RLMObjectBase.__allocating_init() + 32
8   Tezza                         	       0x1019091fc getProperties(_:) + 340 (SchemaDiscovery.swift:218)
9   Tezza                         	       0x101909714 static ObjectUtil.getSwiftProperties(_:) + 144 (SchemaDiscovery.swift:240)
10  Tezza                         	       0x10193dfe4 static RealmSwiftObject._getProperties() + 68 (Object.swift:137)
11  Tezza                         	       0x10193e01c @objc static RealmSwiftObject._getProperties() + 40
12  Tezza                         	       0x10160c79c +[RLMObjectSchema propertiesForClass:isSwift:] + 104 (RLMObjectSchema.mm:222)
13  Tezza                         	       0x10160ac30 +[RLMObjectSchema schemaForObjectClass:] + 1216 (RLMObjectSchema.mm:154)
14  Tezza                         	       0x101816e60 registerClass(objc_class*) + 324 (RLMSchema.mm:118)
15  Tezza                         	       0x101817cc4 __25+[RLMSchema sharedSchema]_block_invoke + 88 (RLMSchema.mm:305)
16  CoreFoundation                	       0x111d2646c __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 16
17  CoreFoundation                	       0x111e4eaec -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 260
18  Tezza                         	       0x101817a38 +[RLMSchema sharedSchema] + 460 (RLMSchema.mm:304)
19  Tezza                         	       0x1017e0284 +[RLMRealm realmWithConfiguration:confinedTo:error:] + 1952 (RLMRealm.mm:458)
20  Tezza                         	       0x1017dfa00 +[RLMRealm realmWithConfiguration:queue:error:] + 172 (RLMRealm.mm:395)
21  Tezza                         	       0x10189f0ec @nonobjc RLMRealm.__allocating_init(configuration:queue:) + 144
22  Tezza                         	       0x10197ae18 Realm.init(queue:) + 220 (Realm.swift:78)
23  Tezza                         	       0x100ce8fb0 static Realm.withRealm<A>(action:) + 208 (Realm+Extension.swift:37)
24  Tezza                         	       0x100d1ef14 RealmServiceType.withRealm<A>(action:) + 220 (RealmServiceType.swift:125)
25  Tezza                         	       0x100d1f928 RealmServiceType.allObjects() + 348 (RealmServiceType.swift:104)
26  Tezza                         	       0x101022654 RealmUserService.activeUser.getter + 132 (RealmUser.swift:56)
27  Tezza                         	       0x100efbd40 AuthProvider.getUser() + 116 (AuthProvider.swift:24)
28  Tezza                         	       0x100f6b198 CleverTapTrackingService.updateUserProfile(withSubscriptionDetails:) + 236 (CleverTapTrackingService.swift:40)
29  Tezza                         	       0x100f6ae18 CleverTapTrackingService.().init() + 312 (CleverTapTrackingService.swift:28)
30  Tezza                         	       0x100f6ac54 CleverTapTrackingService.__allocating_init() + 44
31  Tezza                         	       0x100f6ac14 one-time initialization function for shared + 28 (CleverTapTrackingService.swift:23)
32  libdispatch.dylib             	       0x11a5c3d3c _dispatch_client_callout + 16
33  libdispatch.dylib             	       0x11a5c5530 _dispatch_once_callout + 28
34  Tezza                         	       0x100f6acb0 CleverTapTrackingService.shared.unsafeMutableAddressor + 80 (CleverTapTrackingService.swift:23)
35  Tezza                         	       0x1014573f4 PurchaselyPresenter.().init() + 684 (PurchaselyPresenter.swift:58)
36  Tezza                         	       0x101456f00 PurchaselyPresenter.__allocating_init() + 44
37  Tezza                         	       0x101456ec0 one-time initialization function for shared + 28 (PurchaselyPresenter.swift:47)
38  libdispatch.dylib             	       0x11a5c3d3c _dispatch_client_callout + 16
39  libdispatch.dylib             	       0x11a5c5530 _dispatch_once_callout + 28
40  Tezza                         	       0x101456f5c PurchaselyPresenter.shared.unsafeMutableAddressor + 80 (PurchaselyPresenter.swift:47)
41  Tezza                         	       0x101457118 static PurchaselyPresenter.setup() + 116 (PurchaselyPresenter.swift:50)
42  Tezza                         	       0x10093c5c4 AppDelegate.application(_:didFinishLaunchingWithOptions:) + 416 (AppDelegate.swift:36)
43  Tezza                         	       0x10093e824 @objc AppDelegate.application(_:didFinishLaunchingWithOptions:) + 196
44  UIKitCore                     	       0x130bf9cfc -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 312
45  UIKitCore                     	       0x130bfb15c -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 2788
46  UIKitCore                     	       0x130bffc7c -[UIApplication _runWithMainScene:transitionContext:completion:] + 856
47  UIKitCore                     	       0x1302ad820 -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 148
48  UIKitCore                     	       0x130bfcf70 -[UIApplication _compellApplicationLaunchToCompleteUnconditionally] + 44
49  PreviewsOSSupportUI           	       0x15da5e7f8 -[UVAgentSceneEntryPoint _forceApplicationLaunchCompletion] + 36
50  PreviewsOSSupportUI           	       0x15da5e724 -[UVAgentSceneEntryPoint initWithScene:] + 268
51  UIKitCore                     	       0x13011323c __66+[UIScene _sceneForFBSScene:create:withSession:connectionOptions:]_block_invoke_3 + 84
52  CoreFoundation                	       0x111d2646c __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 16
53  CoreFoundation                	       0x111e4eaec -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 260
54  UIKitCore                     	       0x130112cac +[UIScene _sceneForFBSScene:create:withSession:connectionOptions:] + 648
55  UIKitCore                     	       0x130bfe644 -[UIApplication _connectUISceneFromFBSScene:transitionContext:] + 804
56  UIKitCore                     	       0x130bfe92c -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 356
57  UIKitCore                     	       0x1306edcc4 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 260
58  FrontBoardServices            	       0x11dee8190 -[FBSScene _callOutQueue_didCreateWithTransitionContext:completion:] + 296
59  FrontBoardServices            	       0x11df113d4 __92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke.82 + 224
60  FrontBoardServices            	       0x11def4d6c -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 160
61  FrontBoardServices            	       0x11df1114c __92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke + 284
62  libdispatch.dylib             	       0x11a5c3d3c _dispatch_client_callout + 16
63  libdispatch.dylib             	       0x11a5c7c4c _dispatch_block_invoke_direct + 376
64  FrontBoardServices            	       0x11df34b24 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 44
65  FrontBoardServices            	       0x11df34a00 -[FBSMainRunLoopSerialQueue _targetQueue_performNextIfPossible] + 196
66  FrontBoardServices            	       0x11df34b58 -[FBSMainRunLoopSerialQueue _performNextFromRunLoopSource] + 24
67  CoreFoundation                	       0x111d3df18 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
68  CoreFoundation                	       0x111d3de60 __CFRunLoopDoSource0 + 172
69  CoreFoundation                	       0x111d3d5d0 __CFRunLoopDoSources0 + 232
70  CoreFoundation                	       0x111d37cb8 __CFRunLoopRun + 768
71  CoreFoundation                	       0x111d375a4 CFRunLoopRunSpecific + 572
72  GraphicsServices              	       0x11c9f7ae4 GSEventRunModal + 160
73  UIKitCore                     	       0x130bfd2e4 -[UIApplication _run] + 868
74  UIKitCore                     	       0x130c00f5c UIApplicationMain + 124
75  Tezza                         	       0x100941644 main + 64 (AppDelegate.swift:19)
76  dyld_sim                      	       0x107151544 start_sim + 20
77  dyld                          	       0x106b520e0 start + 2360
```

### Can you reproduce the bug?

Always

### Reproduction Steps

Just try to preview any Swift UI view in Xcode

### Version

10.48.0

### What Atlas Services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

iOS, any version of the simulator

### Build environment

Xcode version: 15.3
Dependency manager and version: ...

**Comment by Jaycyn** [comment]
We may need a bit more info to go on - I have an app running, XCode 15.3, SDK 10.48, iOS, SwiftUI, in-memory realm and it previews and runs correctly.

I then created a fresh project with same config just a simple "Hello World" and a button that writes data to in memory realm, and that works correctly and previews as well.

Perhaps including the code that's crashing would narrow down the cause?

Here's a screenshot of the preview

![Realm](https://github.com/realm/realm-swift/assets/19558530/11e730c9-93ae-4577-8537-9072e5d09724)

**Comment by videolist** [comment]
> We may need a bit more info to go on - I have an app, XCode 15.3, SDK 10.48, iOS, SwiftUI, in-memory realm and it previews and runs correctly.
> 
> I then created a fresh project with same config just a simple "Hello World" and a button that writes data to in memory realm, and that works correctly and previews as well.
> 
> Perhaps including the code that's crashing would narrow down the cause?
> 
> Here's a screenshot of the preview
> 
> ![Realm](https://private-user-images.githubusercontent.com/19558530/312943912-11e730c9-93ae-4577-8537-9072e5d09724.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0NjAxMzQsIm5iZiI6MTcxMDQ1OTgzNCwicGF0aCI6Ii8xOTU1ODUzMC8zMTI5NDM5MTItMTFlNzMwYzktOTNhZS00NTc3LTg1MzctOTA3MmU1ZDA5NzI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDIzNDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyYjQxNWEzODZiM2Y0NmNjZDg0NTMwMjU2NDMyYzY3Njk0N2M0MTUwOGUxMjc2YmUzNjMxOWFlZTBjMmFiNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dXEi9xfSEQ317Gb9xlrcVth1XHJWP0el7sjwY1s-d9s)

HI @Jaycyn

**Comment by videolist** [comment]
Thank you replying @Jaycyn .
Yes I understand it might be specific to our code structure. I will try to narrow it down some more, just wanted to check first if something like that is already on your radar. Will do some more digging.
The frustrating part is that it all works perfectly with Xcode 15.2 and earlier.

**Comment by nirinchev** [comment]
Looking at the stacktrace, it appears it might be related to the existence of a LinkingObjects property in the model.

**Comment by videolist** [comment]
Yes it doesn't like LinkingObjects and the List types, either one leads to a crash.  But I finally found the reason!!!!
This is clearly an Xcode 15.3 bug.
The issue was that for whatever reason my unit tests target had only `Realm` added to its  dependencies but didn't have `RealmSwift` (Frameworks and Libraries panel in General tab for that target's settings).   To solve the issue I had to either add `RealmSwift` dependency or remove the existing `Realm` one. Apparently `RealmSwift` contains definitions for Swift-Only `List` and `LinkingObjects` generic protocols and Xcode wasn't able to resolve them.

Why does it need to build unit tests target for previews, and why does it even use its dependencies for symbol look up, is unwon to me. But this has fixed the issue which was plaguing me for several days.
